src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_STACK_SIZE = 10;
+    private static final int GROW_MULTIPLIER = 2;
+    private Object[] stack;
+    private int size;
+
+    public MateStack() {
+        stack = new Object[DEFAULT_STACK_SIZE];
+        size = 0;
+    }
 
+    public void push(T value) {
+        if (stack.length >= size) {
+            grow();
+        }
```

Comment:
stack.length will almost always be larger than size
```suggestion
        if (size >= stack.length) {
            grow();
        }
```